src/sequelize.js

```diff
@@ -300,6 +300,10 @@ class Sequelize {
       throw new Error('Setting a custom timezone is not supported by SQLite, dates are always returned as UTC. Please remove the custom timezone parameter.');
     }
 
+    if (this.options.dialect === 'ibmi' && this.options.timezone !== '+00:00') {
```

Comment:
maybe out of scope for this PR but I'd prefer if we had a flag on Dialect.supports instead of hardcoding dialects here